src/main/java/com/aws/greengrass/util/Exec.java

```diff
@@ -57,74 +55,44 @@
+    protected Logger logger = staticLogger;
     private static final Consumer<CharSequence> NOP = s -> {
     };
 
     // default directory relative paths are resolved against (i.e. current working directory)
     private static final File userdir = new File(System.getProperty("user.dir"));
 
-    private static final ConcurrentLinkedDeque<Path> paths = new ConcurrentLinkedDeque<>();
-    private static final String PATH = "PATH";
+    protected static final ConcurrentLinkedDeque<Path> paths = new ConcurrentLinkedDeque<>();
+    protected static final String PATH = "PATH";
     private static final Map<String, String> defaultEnvironment = new ConcurrentHashMap<>();
+    protected final Map<String, String> environment = new HashMap<>(defaultEnvironment);
 
     static {
         addPathEntries(System.getenv(PATH));
+        computePathString();
         defaultEnvironment.put("JAVA_HOME", System.getProperty("java.home"));
         defaultEnvironment.put("HOME", System.getProperty("user.home"));
-        try {
-            if (!isWindows) {
-                // This bit is gross: under some circumstances (like IDEs launched from the
-                // macos Dock) the PATH environment variable doesn't match the path one expects
-                // after the .profile script is executed.  Fire up a login shell, then grab it's
-                // path variable, but without using Exec shorthands to avoid initialization
-                // order paradoxes.
-                Process hack =
-                        Runtime.getRuntime().exec(new String[]{"sh", "-c", "echo 'echo $PATH' | grep -E ':[^ ]'"});
-                StringBuilder path = new StringBuilder();
-
-                Thread bg = new Thread(() -> {
-                    try (InputStream in = hack.getInputStream()) {
-                        for (int c = in.read(); c >= 0; c = in.read()) {
-                            path.append((char) c);
-                        }
-                    } catch (Throwable ignore) {
-                    }
-                });
-                bg.start();
-                bg.join(2000);
-                addPathEntries(path.toString().trim());
-                // Ensure some level of sanity
-                ensurePresent("/bin", "/usr/bin", "/sbin", "/usr/sbin");
-            }
-        } catch (Throwable ex) {
-            staticLogger.atError().log("Error while initializing PATH", ex);
-        }
-        computePathString();
     }
 
-    private final AtomicBoolean isClosed = new AtomicBoolean(false);
-    Process process;
-    IntConsumer whenDone;
-    Consumer<CharSequence> stdout = NOP;
-    Consumer<CharSequence> stderr = NOP;
-    AtomicInteger numberOfCopiers;
-    private final Map<String, String> environment = new HashMap<>(defaultEnvironment);
-    private String[] cmds;
+    protected final AtomicBoolean isClosed = new AtomicBoolean(false);
+    private Process process;
```

Comment:
why is process private? That should be protected shouldn't it?

Reply:
Right now the Exec child classes just implement `createProcess`. The super class uses the returned `Process` object to do waitFor, destroy, etc. The platform-specific code will be in the `Process` class like `WindowsRunasProcess` for Windows.

Reply:
The implementation of `close` will be platform specific, and close needs access to the Process

Reply:
I see. The PidProcess thing can't shutdown gracefully on windows

Reply:
Not likely, no. It should be override-able